go/libraries/doltcore/merge/violations_fk_prolly.go

```diff
@@ -329,7 +333,7 @@ func createCVsForPartialKeyMatches(
 }
 
 func makePartialKey(kb *val.TupleBuilder, tags []uint64, idxSch schema.Index, tblSch schema.Schema, k, v val.Tuple, pool pool.BuffPool) (val.Tuple, bool) {
-	if idxSch.Name() != "" {
+	if idxSch.Name() != "" && len(idxSch.IndexedColumnTags()) <= len(tags) {
```

Comment:
this one could use a comment too, an example of how we'd hit this condition.

Reply:
We hit this condition when the parent's index `idxSch` is longer than the partial key we have `k` with length `len(tags)`, so we panic trying to build a key that's too long.